### PR TITLE
site: Add instructions for Linux OS systemd-resolved to ingress-dns documentation

### LINKED ISSUE #18727: Ingress DNS documentation lacks instruction for Linux OS systemd-resolved (relevant to Ubuntu)
The [Ingress DNS documentation](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/) has instruction for adding the `minikube ip` as a DNS server ("Installlation", step 3). Under the Linux tab, it provides instructions to do so for systems using `resolvconf` and `Network Manager`, but not [`systemd-resolved`](https://manpages.ubuntu.com/manpages/bionic/man8/systemd-resolved.service.8.html). This is relevant because it seems to be used by Ubuntu>=22.04 (`/etc/resolv.conf` starts with `# This is /run/systemd/resolve/stub-resolv.conf managed by man:systemd-resolved(8).`)

On systems using `systemd-resolved`, the minikube IP can be added by running:

```shell
sudo mkdir /etc/systemd/resolved.conf.d
sudo tee /etc/systemd/resolved.conf.d/minikube.conf << EOF
[Resolve]
DNS=$(minikube ip)
Domains=~test
EOF
sudo systemctl restart systemd-resolved
```

I propose to create a PR to add this to the documentation.

### PR DESCRIPTION
**Pull request by gpelouze**
Fixes #18727

This PR adds instructions to setup the ingress DNS on Linux systems using `systemd-resolved`. This is relevant because Ubuntu>=22.04 uses this resolver.

### CODE REVIEW

File: site/content/en/docs/handbook/addons/ingress-dns.md [hunk 1]
```diff
@@ -117,6 +117,22 @@ cat /etc/resolv.conf | grep nameserver
 nameserver 127.0.0.1
 ```
 
+## Linux OS with systemd-resolved
+
+Your Linux OS likely uses `systemd-resolved` if the first line of `/etc/resolv.conf` contains `This is /run/systemd/resolve/stub-resolv.conf managed by man:systemd-resolved(8)`.
+
+Run the following commands to add the Minikube DNS for `.test` domains:
```
Comment (medyagh) [inline site/content/en/docs/handbook/addons/ingress-dns.md:124]: minikube should be lowercase

File: site/content/en/docs/handbook/addons/ingress-dns.md [hunk 2]
```diff
@@ -117,6 +117,22 @@ cat /etc/resolv.conf | grep nameserver
...
+
+```bash
+sudo mkdir /etc/systemd/resolved.conf.d
```
Comment (medyagh) [inline site/content/en/docs/handbook/addons/ingress-dns.md:127]: let do -p incase it exsits

```sudo mkdir -p /etc/systemd/resolved.conf.d
```

**Review by medyagh — CHANGES_REQUESTED** [review]
thanks for this PR do u mind pasting an example of this Before this instructions and does it apply to all Linux users ? or only ubuntu?

**Comment by Dmarcotrigiano** [comment]
When following these docs, I can now correctly connect to my services from my host machine at their <service>.test domain, but API calls to other services such as below are now failing.
```
https://api.stripe.com/v1/products\": dial tcp: lookup api.stripe.com on 10.96.0.10:53: no such host
```

**Pull request merged**